server/src/test/java/com/microsoft/java/bs/core/internal/server/LifecycleServiceIntegrationTest.java

```diff
@@ -0,0 +1,137 @@
+package com.microsoft.java.bs.core.internal.server;
+
+import ch.epfl.scala.bsp4j.BuildClientCapabilities;
+import ch.epfl.scala.bsp4j.InitializeBuildParams;
+import ch.epfl.scala.bsp4j.MessageType;
+import ch.epfl.scala.bsp4j.ShowMessageParams;
+import com.microsoft.java.bs.core.internal.model.Preferences;
+import com.microsoft.java.bs.gradle.model.SupportedLanguages;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class LifecycleServiceIntegrationTest extends IntegrationTest {
+
+  private static InitializeBuildParams getInitializedBuildParamsWithJdks(
+      String projectDir,
+      String jdkVersion,
+      String gradleJavaVersionPath
+  ) {
+    File root = Paths.get(
+        System.getProperty("user.dir"),
+        "..",
+        "testProjects",
+        projectDir).toFile();
+
+    BuildClientCapabilities capabilities =
+        new BuildClientCapabilities(SupportedLanguages.allBspNames);
+    final InitializeBuildParams initParams = new InitializeBuildParams(
+        "test-client",
+        "0.1.0",
+        "0.1.0",
+        root.toURI().toString(),
+        capabilities
+    );
```

Comment:
Looks like we can directly get the initParams by `IntegrationTest.getInitializeBuildParams(projectDir)`, which can remove duplicated code

Reply:
Fixed in https://github.com/microsoft/build-server-for-gradle/pull/165/commits/24fdf827bf77e943ea93778a3b3fdd7b78a9b94c